astropy/stats/sigma_clipping.py

```diff
@@ -879,6 +881,287 @@ def sigma_clip(
+        used. The default is `None`.
+
+    sigma_upper : float or None, optional
+        The number of standard deviations to use as the upper bound for
+        the clipping limit. If `None` then the value of ``sigma`` is
+        used. The default is `None`.
+
+    maxiters : int or None, optional
+        The maximum number of sigma-clipping iterations to perform or
+        `None` to clip until convergence is achieved (i.e., iterate
+        until the last iteration clips nothing). If convergence is
+        achieved prior to ``maxiters`` iterations, the clipping
+        iterations will stop. The default is 5.
+
+    cenfunc : {'median', 'mean'} or callable, optional
+        The statistic or callable function/object used to compute
+        the center value for the clipping. If using a callable
+        function/object and the ``axis`` keyword is used, then it must
+        be able to ignore NaNs (e.g., `numpy.nanmean`) and it must have
+        an ``axis`` keyword to return an array with axis dimension(s)
+        removed. The default is ``'median'``.
+
+    stdfunc : {'std', 'mad_std'} or callable, optional
+        The statistic or callable function/object used to compute the
+        standard deviation about the center value. If using a callable
+        function/object and the ``axis`` keyword is used, then it must
+        be able to ignore NaNs (e.g., `numpy.nanstd`) and it must have
+        an ``axis`` keyword to return an array with axis dimension(s)
+        removed. The default is ``'std'``.
+
+    axis : None or int or tuple of int, optional
+        The axis or axes along which to sigma clip the data. If `None`,
+        then the flattened data will be used. ``axis`` is passed to the
+        ``cenfunc`` and ``stdfunc``. The default is `None`.
+
+    grow : float or `False`, optional
+        Radius within which to mask the neighbouring pixels of those
+        that fall outwith the clipping limits (only applied along
+        ``axis``, if specified). As an example, for a 2D image a value
+        of 1 will mask the nearest pixels in a cross pattern around each
+        deviant pixel, while 1.5 will also reject the nearest diagonal
+        neighbours and so on.
+
+    Notes
+    -----
+    The best performance will typically be obtained by setting
+    ``cenfunc`` and ``stdfunc`` to one of the built-in functions
+    specified as a string. If one of the options is set to a string
+    while the other has a custom callable, you may in some cases
+    see better performance if you have the `bottleneck`_ package
+    installed. To preserve accuracy, bottleneck is only used for float64
+    computations.
+
+    .. _bottleneck:  https://github.com/pydata/bottleneck
+
+    See Also
+    --------
+    sigma_clipped_stats, SigmaClip, sigma_clip
+    """
+
+    def __init__(
+        self,
+        data: ArrayLike,
+        mask: NDArray | None = None,
+        mask_value: float | None = None,
+        sigma: float | None = 3.0,
```

Comment:
what does it mean to pass `sigma=None` ? This looks like it could be a misunderstanding of the meaning of this type-annotation: `| None` is not synonymous with "has a default value" (though the name of `typing.Optional` may be the source of confusion), but really means "this parameter can be `None`".

If passing `sigma=None` is really not supported, as I suspect, the annotation should be
```suggestion
        sigma: float = 3.0,
```

Reply:
I'm starting to think the confusion may have originated from #16562, where I *think* some similar looking mistakes may have been made. I'll try to fix those in a separate PR.